src/wallet/wallet.cpp

```diff
@@ -2736,6 +2736,33 @@ bool CWalletTx::IsTrusted() const
     return true;
 }
 
+bool CWalletTx::IsTrusted(bool spendZeroConfChangeIn) const
+{
+    // Quick answer in most cases
+    if (!CheckFinal())
+        return false;
+    int nDepth = GetDepthInMainChain();
+    if (nDepth >= 1)
+        return true;
+    if (nDepth < 0)
+        return false;
+    if (!spendZeroConfChangeIn || !IsFromMe(ISMINE_ALL)) // using wtx's cached debit
+        return false;
+
+    // Trusted if all inputs are from us and are in the mempool:
+    BOOST_FOREACH(const CTxIn& txin, GetVin())
+    {
+        // Transactions not sent by us: not trusted
+        const CWalletObjBase* parent = pwallet->GetWalletTx(txin.prevout.hash);
+        if (parent == NULL)
+            return false;
+        const CTxOut& parentOut = parent->GetVout()[txin.prevout.n];
+        if (pwallet->IsMine(parentOut) != ISMINE_SPENDABLE)
+            return false;
+    }
+    return true;
+}
+
```

Comment:
`bool CWalletTx::IsTrusted(bool spendZeroConfChangeIn) const ` is the copy of `bool CWalletTx::IsTrusted() const` with the exception of `spendZeroConfChangeIn `.
Can you write a single function with bool input parameter set to bSpendZeroConfChange?